src/components/CartList/CartList.jsx

```diff
@@ -1,27 +1,135 @@
-import React from "react";
+import React, { useEffect } from "react";
 import styles from "./CartList.module.css";
 import { CartCard } from "../Card/CartCard";
 import { useCart } from "../../context/cart-context";
 import { useWishlist } from "../../context/wishlist-context";
+import { useHttp } from "../../hooks/useHttp";
+import { useAuth } from "../../context/auth-context";
+import { deleteFromCart } from "../../helpers/deleteFromCart";
+import { ToastContainer, toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import { changeInCart } from "../../helpers/changeInCart";
+import { moveItem } from "../../helpers/moveItem";
 export const CartList = () => {
   const { cart, cartDispatch } = useCart();
-  const { wishlistDispatch } = useWishlist();
+  const { wishlistDispatch, wishlist } = useWishlist();
+  const { authState } = useAuth();
+  const { sendRequest, data, error, status } = useHttp(deleteFromCart);
+  const {
+    sendRequest: changeRequest,
+    data: changeData,
+    error: changeError,
+    status: changeStatus,
+  } = useHttp(changeInCart);
+  const {
+    sendRequest: moveRequest,
+    data: moveData,
+    error: moveError,
+    status: moveStatus,
+  } = useHttp(moveItem);
   const onMoveFromCartToWIshlist = (product) => {
-    wishlistDispatch({ type: "ADD", payload: product });
-    cartDispatch({ type: "DELETE", payload: product });
+    moveRequest({token: authState.token, product, action: "CART_TO_WISHLIST"});
   };
+  const onRemoveFromCart = (product) => {
+    sendRequest({ product, token: authState.token });
+  };
+  const changeItemQuantity = (product, type) => {
+    changeRequest({ product, token: authState.token, type });
+  };
+  const isInWishlist = (product) => {
+    return wishlist.find((item) => item._id === product._id);
+  };
+  useEffect(() => {
+    if (status === "completed") {
+      cartDispatch({ type: "REMOVE", payload: data });
+      toast.success("Item removed from cart!", {
+        position: "top-right",
+        autoClose: 1000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: false,
+        draggable: true,
+        progress: undefined,
+      });
```

Comment:
you could have extracted in another function to make it look clean everywhere you are using toastify